### ISSUE TITLE
Multi-progress bars don't work

### ISSUE THREAD

**Issue by arxanas** [mitsuhiko/indicatif#304]
Hi, I haven't been able to get multi-progress bars to render at all. The regular single progress bars work fine.

Here's me trying the example at https://github.com/mitsuhiko/indicatif/blob/79ef74a4e9f74c8d135399bd2ebb864c5eb81acd/examples/morebars.rs.

Expected behavior: some kind of output appears when updating progress bars registered with a `MultiProgress`.
Actual behavior: no output appears.

[![asciicast](https://asciinema.org/a/LdCJiFhWc6MsWFqpQFH6yz6Fe.svg)](https://asciinema.org/a/LdCJiFhWc6MsWFqpQFH6yz6Fe)

Tried with Terminal.app, iTerm2, and the VS Code integrated terminal.

OS information:

```
$ echo "$TERM"
xterm-256color
$ uname -a
Darwin waleedkhan-mbp 20.5.0 Darwin Kernel Version 20.5.0: Sat May  8 05:10:33 PDT 2021; root:xnu-7195.121.3~9/RELEASE_X86_64 x86_64
```

Cargo.toml:

```
[package]
name = "indicatif-test"
version = "0.1.0"
edition = "2018"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
indicatif = "0.16.2"
# This file is automatically @generated by Cargo.
# It is not intended for manual editing.
version = 3
```

Cargo.lock:

```
[[package]]
name = "console"
version = "0.14.1"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "3993e6445baa160675931ec041a5e03ca84b9c6e32a056150d3aa2bdda0a1f45"
dependencies = [
 "encode_unicode",
 "lazy_static",
 "libc",
 "terminal_size",
 "winapi",
]

[[package]]
name = "encode_unicode"
version = "0.3.6"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "a357d28ed41a50f9c765dbfe56cbc04a64e53e5fc58ba79fbc34c10ef3df831f"

[[package]]
name = "indicatif"
version = "0.16.2"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "2d207dc617c7a380ab07ff572a6e52fa202a2a8f355860ac9c38e23f8196be1b"
dependencies = [
 "console",
 "lazy_static",
 "number_prefix",
 "regex",
]

[[package]]
name = "indicatif-test"
version = "0.1.0"
dependencies = [
 "indicatif",
]

[[package]]
name = "lazy_static"
version = "1.4.0"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "e2abad23fbc42b3700f2f279844dc832adb2b2eb069b2df918f455c4e18cc646"

[[package]]
name = "libc"
version = "0.2.98"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "320cfe77175da3a483efed4bc0adc1968ca050b098ce4f2f1c13a56626128790"

[[package]]
name = "number_prefix"
version = "0.4.0"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "830b246a0e5f20af87141b25c173cd1b609bd7779a4617d6ec582abaf90870f3"

[[package]]
name = "regex"
version = "1.5.4"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "d07a8629359eb56f1e2fb1652bb04212c072a87ba68546a04065d525673ac461"
dependencies = [
 "regex-syntax",
]

[[package]]
name = "regex-syntax"
version = "0.6.25"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "f497285884f3fcff424ffc933e56d7cbca511def0c9831a7f9b5f6153e3cc89b"

[[package]]
name = "terminal_size"
version = "0.1.17"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "633c1a546cee861a1a6d0dc69ebeca693bf4296661ba7852b9d21d159e0506df"
dependencies = [
 "libc",
 "winapi",
]

[[package]]
name = "winapi"
version = "0.3.9"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "5c839a674fcd7a98952e593242ea400abe93992746761e38641405d28b00f419"
dependencies = [
 "winapi-i686-pc-windows-gnu",
 "winapi-x86_64-pc-windows-gnu",
]

[[package]]
name = "winapi-i686-pc-windows-gnu"
version = "0.4.0"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "ac3b87c63620426dd9b991e5ce0329eff545bccbbb34f3be09ff6fb6ab51b7b6"

[[package]]
name = "winapi-x86_64-pc-windows-gnu"
version = "0.4.0"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "712e227841d057c1ee1cd2fb22fa7e5a5461ae8e48fa2ca79ec42cfc1931183f"
```

**Comment by arxanas** [comment]
Okay, it looks like the example in `examples/` doesn't work under indicatif 0.16.2. Would it be possible to put a pre-release of the current version on crates.io, so that I can depend on it directly? It looks like the `MultiProgress` API has changed meaningfully, and I don't want to have to change it again later.

**Comment by djc** [comment]
I've published 0.17.0-beta.1.

**Comment by gabrielfalcao** [comment]
> I've published 0.17.0-beta.1.

Spent more than an hour running this issue when using tokio with async main function. Finally found this and it worked.

**Comment by lifegpc** [comment]
I got same issue in 1.16.2
